platforms/core-runtime/service-registry-builder/src/main/java/org/gradle/internal/service/ServiceRegistryBuilder.java

```diff
@@ -31,25 +37,62 @@ public class ServiceRegistryBuilder {
     private ServiceRegistryBuilder() {
     }
 
+    /**
+     * Creates a new builder.
+     */
     public static ServiceRegistryBuilder builder() {
         return new ServiceRegistryBuilder();
     }
 
+    /**
+     * Sets the display name to be used by the service registry.
+     * <p>
+     * The display name is used for debugging and in the errors messages.
+     * The errors are not user-facing and oriented at troubleshooting.
```

Comment:
I agree this sounds strange, and I'm open to better wording.

But I think it's important to highlight that this display name is more similar to the implementations of `toString()` for internal classes, rather than something we expect the user to see. The user might see this in an error message, but only if something goes very wrong and Gradle is broken in a bad way. Similar to how we sometimes throw `IllegalStateException`, not expecting users to ever see this.

Reply:
I decided that this paragraph is not appropriate as part of the display name javadoc. I simplified the text.